ignite/contrib/checkpoint.py

```diff
@@ -0,0 +1,131 @@
+import os
+import tempfile
+from collections import namedtuple
+
+import torch
+
+
+class ModelCheckpoint:
+    """ Foo bar
+
+    Parameters
+    ----------
+    dirname : str
+        Directory path where objects will be saved
+    filename_prefix : str
+        Prefix for the filenames to which objects will be saved.
+        Each object is saved under '{filename_prefix}_{name}_{step_number}.pth'
+        where {step_number} is epoch or iteration number.
+    iteration_interval : int, optional
+        if not None, model will be saved to disk every 'iteration_interval' iterations
+    epoch_interval : int, optional
+        if not None, model will be saved to disk every 'epoch_interval' epochs
+    score_function : Callable, optional
```

Comment:
It wasn't clear to me from the docstrings how the priority is calculated and balanced between `iteration_interval`, `epoch_interval` and the `score_function`. For example, if I want to checkpoint when the lowest training score is hit, and also every 5th iteration I don't think this is possible, as the function will terminate early if it's not a 5th iteration. 

I'm ok with this not being possible, but it should be clear in the docstring and we should also raise an error if more than one of `iteration_interval`, `epoch_interval` or `score_function` are set?

Reply:
I've updated the docstring.

I have also decided to combine the `iteration_interval` and `epoch_interval`. With previous solution you had to **both** attach the handler to the correct event, and provide the correct argument. I believe it is nicer now.